src/package.test.ts

```diff
@@ -2,40 +2,426 @@ import fs from 'fs';
+            name: 'some-package',
+            version: new SemVer('1.0.0'),
+          }),
+        });
+      when(jest.spyOn(repoModule, 'hasChangesInDirectorySinceGitTag'))
+        .calledWith('/path/to/project', '/path/to/package', 'v1.0.0')
+        .mockResolvedValue(true);
+
+      const promiseForPkg = readMonorepoRootPackage({
+        packageDirectoryPath: '/path/to/package',
+        projectDirectoryPath: '/path/to/project',
+        projectTagNames: ['some-tag'],
+      });
+
+      await expect(promiseForPkg).rejects.toThrow(
+        new Error(
+          'The package some-package has no Git tag for its current version 1.0.0 (expected "v1.0.0"), so this tool is unable to determine whether it should be included in this release. You will need to create a tag for this package in order to proceed.',
+        ),
+      );
+    });
+
+    it("returns the fact that the package has been changed since its latest release, if a tag matching the current version exists and changes have been made to the package's directory since the tag", async () => {
+      jest
+        .spyOn(packageManifestModule, 'readPackageManifest')
+        .mockResolvedValue({
+          unvalidated: {},
+          validated: buildMockManifest({
+            version: new SemVer('1.0.0'),
+          }),
+        });
+      when(jest.spyOn(repoModule, 'hasChangesInDirectorySinceGitTag'))
+        .calledWith('/path/to/project', '/path/to/package', 'v1.0.0')
+        .mockResolvedValue(true);
+
+      const pkg = await readMonorepoRootPackage({
+        packageDirectoryPath: '/path/to/package',
+        projectDirectoryPath: '/path/to/project',
+        projectTagNames: ['v1.0.0'],
+      });
+
+      expect(pkg).toMatchObject({
+        hasChangesSinceLatestRelease: true,
+      });
+    });
+
+    it("returns the fact that the package has not been changed since its latest release if a tag matching the current version exists, and changes have not been made to the package's directory since the tag", async () => {
+      jest
+        .spyOn(packageManifestModule, 'readPackageManifest')
+        .mockResolvedValue({
+          unvalidated: {},
+          validated: buildMockManifest({
+            version: new SemVer('1.0.0'),
+          }),
+        });
+      when(jest.spyOn(repoModule, 'hasChangesInDirectorySinceGitTag'))
+        .calledWith('/path/to/project', '/path/to/package', 'v1.0.0')
+        .mockResolvedValue(false);
+
+      const pkg = await readMonorepoRootPackage({
+        packageDirectoryPath: '/path/to/package',
+        projectDirectoryPath: '/path/to/project',
+        projectTagNames: ['v1.0.0'],
+      });
+
+      expect(pkg).toMatchObject({
+        hasChangesSinceLatestRelease: false,
+      });
+    });
+
+    it("returns the fact that the package has been changed since its latest release, if a tag matching 'v' + the current version exists and changes have been made to the package's directory since the tag", async () => {
```

Comment:
This test looks identical to the one two before it :thinking:

Reply:
Ah. I only spotted one duplicate, so I removed that. As for the other tests, they follow a "this should be true if this is the case" / "this should not be true if this is not the case" pattern, so perhaps that was difficult to see. I renamed the tests to make that distinction more clear.

Reply:
Huh, it seems that both duplicate tests I saw are removed in that commit. The re-wording looks good too though!